client/src/components/mobile/ResourceDetailMobile.js

```diff
@@ -0,0 +1,471 @@
+import moment from 'moment';
+
+import ResourceDetail from '../../pages/ResourceDetail';
+import ResourcesBreadcrumb from '../ResourcesBreadcrumb';
+import SaveButton from '../SaveButton';
+import useWindowDimensions from '../../utils/mobile';
+import { deleteResource, getResourceByID } from '../../utils/api';
+import {
+  saveResource,
+  deleteSavedResource,
+  getSavedResources,
+} from '../../utils/auth';
+import determineStockPhoto from '../../utils/determineStockPhoto';
+
+import '../../css_mobile/ResourceDetail.css';
+
+type Props = {
+  authed: Boolean,
+  authRoleIsEquivalentTo: String => Boolean,
+  match: {
+    params: {
+      id: any,
+    },
+  },
+};
+
+const ResourceDetailMobile = (props: Props) => {
+  const { authed, authRoleIsEquivalentTo } = props;
+
+  const resourceId = props.match.params.id;
+  const isMobile = useWindowDimensions()[1];
+
+  /* SETUP START */
+
+  // ResourceDetail page will get all this stuff prior to this
+  // being loaded but for right now everything will be fetched.
+  // by this component.
+  const [name, setName] = useState(null);
+  const [category, setCategory] = useState(null);
+  const [subcategory, setSubcategory] = useState(null);
+  const [description, setDescription] = useState(null);
+  const [website, setWebsite] = useState(null);
+  const [email, setEmail] = useState(null);
+  const [phone, setPhone] = useState([]);
+  const [address, setAddress] = useState(null);
+  const [addressLine2, setAddressLine2] = useState(null);
+  const [aptUnitSuite, setAptUnitSuite] = useState('');
+  const [city, setCity] = useState('');
+  const [state, setState] = useState('');
+  const [zip, setZip] = useState('');
+  const [languages, setLanguages] = useState([]);
+  const [requiredDocuments, setRequiredDocuments] = useState(null);
+  const [cost, setCost] = useState(null);
+  const [internalNotes, setInternalNotes] = useState([]);
+  const [hours, setHours] = useState(null);
+  const [recommendation, setRecommendation] = useState(0);
+  const [image, setImage] = useState(null);
+
+  const [resourceExists, setResourceExists] = useState(true);
+
+  const [lat, setLat] = useState(0);
+  const [lng, setLng] = useState(0);
+  const [distFromResource, setDistFromResource] = useState(null);
+
+  const [eligibility, setEligibility] = useState('');
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const [isSaved, setIsSaved] = useState(false);
+  const [isWithinOperationHours, setIsWithinOperationHours] = useState(null);
+
+  // componentDidMount
+  useEffect(() => {
+    async function loadResource() {
+      const response = await getResourceByID(resourceId, true);
+
+      if (response) {
+        const { result } = response;
+
+        setImage(
+          result.image && result.image !== ''
+            ? result.image
+            : determineStockPhoto(result.category, result.subcategory),
+        );
+
+        setCategory(result.category);
+        setSubcategory(result.subcategory);
```

Comment:
The categories and subcategories are lists of strings, so doing this may mess up some code where it is used.  For example, the breadcrumb - to get around this you can just take the first entry in both category and subcategory